File: Set Matrix Zeroes.cpp
class Solution {
    public:
        void setZeroes(vector<vector<int>>& matrix)
        {
            int rowSize = matrix.size();
            int colSize = matrix[0].size();
            bool m[rowSize];
            bool n[colSize];
            fill(m, m+rowSize, false);
            fill(n, n+colSize, false);
            for(int i=0; i<rowSize; i++)
            {
                for(int j=0; j<colSize; j++)
                {
                    if(matrix[i][j]==0)
                    {
                        m[i]=true;
                        n[j]=true;
                    }
                }
            }
            for(int i=0; i<rowSize; i++)
            {
                for(int j=0; j<colSize; j++)
                {
                    if(m[i]==true || n[j]==true)
                    {
                       matrix[i][j]=0;
                    }
                }
            }
        }
    };

File: Set Matrix Zeroes.py
class Solution(object):
    def setZeroes(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: None Do not return anything, modify matrix in-place instead.
        """
        rowSize=len(matrix)
        colSize=len(matrix[0])
        m = [False] * rowSize;
        n = [False] * colSize;
        for i in range(rowSize):
            for j in range(colSize):
                if matrix[i][j] == 0:
                    m[i] = True;
                    n[j] = True;

        for i in range(rowSize):
            for j in range(colSize):
                if m[i] or n[j]:
                    matrix[i][j] = 0